src/main/java/com/godaddy/sonar/ruby/RubySensor.java

```diff
@@ -27,30 +29,33 @@ public RubySensor(Settings settings, FileSystem fileSystem) {
         this.fileSystem = fileSystem;
     }
 
-    @Override
     public boolean shouldExecuteOnProject(Project project) {
         // This sensor is executed only when there are Ruby files
         return fileSystem.hasFiles(fileSystem.predicates().hasLanguage(Ruby.KEY));
     }
 
-    public void analyse(Project project, SensorContext context) {
-        computeBaseMetrics(context, project);
+    public void describe(SensorDescriptor descriptor){
+        descriptor.onlyOnLanguage(Ruby.KEY).name("Ruby Sensor").onlyOnFileType(InputFile.Type.MAIN);
+    }
+
+    public void execute(SensorContext context) {
+        computeBaseMetrics(context);
     }
 
-    protected void computeBaseMetrics(SensorContext sensorContext, Project project) {
+    protected void computeBaseMetrics(SensorContext sensorContext) {
```

Comment:
add function docstring for non-API function. See[1]
[1] https://github.com/shakedlokits/ruby-sonar-plugin/blob/6.7-dev/src/main/java/com/godaddy/sonar/ruby/metricfu/MetricfuComplexitySensor.java#L83

Reply:
Added.